### PR TITLE
improve logs and metrics

### LINKED ISSUE #76: improve logging & metrics
### Description

As we are developing RPCh, it's critical that we have a consistent and flexible logging system.
Currently, we rely on `common.utils.createLogger` which aims to provide three levels of logging.

1. `.log` is used for important logical milestones that make sense to log
2. `.logVerbose` verbose logging
3. `.logError` logging of errors

Additionally, it is expected that the developer which uses `createLogger` also uses the right paradigm, for example:
`createLogger(["funding-service", "access-token", "service"])` which would result in a log ID of `funding-service:access-token:service <some error>`.

Most importantly, the `createLogger` function does not provide the functionality for allowing us to utilize metrics, as it requires the metric analyzing system to use tools such as `regexp` to match with various logs and thus create metrics.

In this `epic` we aim on achieving the following:

1. adjust `createLogger` to require the developer to indicate the package in which it's used, example: `createLogger("funding-service" ..`
2. as the last argument of the log, allow the pass of a generic object which can take all sorts of variables in, this should be typed correctly via typescript

Example interface:

```typescript
type InterestingThings = Partial<{
  peerId: string,
  [key: any]: any
}>
type CreateLogger = (app: string, path: string[]) => (...args: any, forMetrics: InterestingThings) => { ... }
```

### Relevant issues

- [x] sync with RPCh on the proposal
- [x] sync with devops team on whether this metric object is useful at all
- [x] implement CreateLoggerFactory
- [x] update RPCh commons
- [x] update RPCh sdk
- [x] update RPCh ethers
- [x] update RPCh exit-node
- [x] update RPCh funding-service

### Definition of DONE

Once we have implemented a consistent logging system throughout the codebase which can be easily utilized by a metrics software.

### PR DESCRIPTION
**Pull request by nionis**
Fixes #76

### CODE REVIEW

File: packages/common/src/cache.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default class Cache {
     };
 
     if (segmentEntry.segments.find((s) => s.segmentNr === segment.segmentNr)) {
-      logVerbose(
+      log.verbose(
         "dropping segment, already exists",
         segment.msgId,
         segment.segmentNr
```
Comment (nionis) [inline packages/common/src/cache.ts:51]: lets add metric here

File: packages/common/src/cache.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export default class Cache {
 
     segmentEntry.segments = [...segmentEntry.segments, segment];
     this.segments.set(segment.msgId, segmentEntry);
-    logVerbose("stored new segment for message ID", segment.msgId);
+    log.verbose("stored new segment for message ID", segment.msgId);
```
Comment (nionis) [inline packages/common/src/cache.ts:58]: lets add metric here

File: packages/common/src/cache.ts [hunk 3]
```diff
@@ -82,11 +90,11 @@ export default class Cache {
   public removeExpired(timeout: number): void {
     const now = new Date();
 
-    logVerbose("total number of segments", this.segments.size);
+    log.verbose("total number of segments", this.segments.size);
```
Comment (nionis) [inline packages/common/src/cache.ts:93]: lets add metric here

File: packages/ethers/src/index.ts [hunk 4]
```diff
@@ -1,18 +1,17 @@
 import { JsonRpcProvider } from "@ethersproject/providers";
 import { deepCopy } from "@ethersproject/properties";
 import SDK, { type HoprSdkTempOps } from "rpch-sdk";
-import { utils } from "rpch-common";
-import { parseResponse, getResult } from "./utils";
+import { parseResponse, getResult, createLogger } from "./utils";
 
-const { logVerbose, logError } = utils.createLogger(["ethers"]);
+const log = createLogger(["ethers"]);
```
Comment (nionis) [inline packages/ethers/src/index.ts:6]: this doesnt need a path

**Review by nionis — COMMENTED** [review]
looks good

File: packages/sdk/src/index.ts [hunk 5]
```diff
@@ -133,15 +133,20 @@ export default class SDK {
       "success"
     );
     this.requestCache.removeRequest(matchingRequest.request);
```
Comment (nionis) [inline packages/sdk/src/index.ts:135]: can we add a "response time" log here? by comparing the time the request was created with the time the response was found

Comment (ItsLuized) [reply packages/sdk/src/index.ts:143]: Good idea! Next commit has the addition.

**Review by ItsLuized — APPROVED** [review]

**Pull request merged**